libnavui-base/src/main/java/com/mapbox/navigation/ui/base/util/ViewUtils.kt

```diff
@@ -0,0 +1,59 @@
+package com.mapbox.navigation.ui.base.util
+
+import android.graphics.Bitmap
+import android.graphics.Canvas
+import android.util.Base64
+import android.view.View
+import android.view.ViewGroup
+import com.mapbox.navigation.core.MapboxNavigation
+import java.io.ByteArrayOutputStream
+import kotlin.math.min
+import kotlin.math.roundToInt
+
+/**
+ * Provides utilities to capture and encode screenshots, accepted by [MapboxNavigation.postUserFeedback].
+ */
+object ViewUtils {
+
+    /**
+     * Captures screenshot of a [View] and returns it as a [Bitmap].
+     * Warning: not suitable for Mapbox [MapView]. Use its [snapshot] function instead.
+     *
+     * @return captured [Bitmap]
+     */
+    @JvmStatic
+    fun View.captureBitmap(): Bitmap {
+        val bitmap = if (isLaidOut) {
+            Bitmap.createBitmap(width, height, Bitmap.Config.ARGB_8888)
+        } else {
+            val measureSpec = View.MeasureSpec.makeMeasureSpec(0, View.MeasureSpec.UNSPECIFIED)
+            measure(measureSpec, measureSpec)
+            layout(left, top, right, bottom)
+            Bitmap.createBitmap(measuredWidth, measuredHeight, Bitmap.Config.ARGB_8888)
```

Comment:
Not sure about this piece. While it works in my tests, maybe it would be safer to return `null` if the view is not yet ready, like `MapView.snapshot` does.